antiplag/models.py

```diff
@@ -47,24 +47,39 @@ def create_and_process_text(cls, submission=None, text_raw=None, file=None, lang
         type = cls.DocumentType.FILE if file else cls.DocumentType.TEXT
         text = cls.process_raw_text(text_raw)
 
-        document = cls(file=file, submission=submission, text=text, text_raw=text_raw, type=type, language=language)
-        document = document.save() if save else document
-        # TODO: Add document to elastic
+        language = cls.detect_language(text)
+
+        document = cls.objects.create(
+            file=file,
+            submission=submission,
+            text=text,
+            text_raw=text_raw,
+            type=type,
+            language=language
+        )
+        document = document.save()
+
         return document
 
     def __str__(self):
-        return f"document-{self.id}-{self.type.label}"
+        return f"document-{self.id}-{self.type}"
```

Comment:
`label` attribute is usually for human readable representation of `TextChoices`.

Reply:
Changed back to `.label` ✅